### ISSUE TITLE
Feature Request: topic is read-only by default when not logged in

### ISSUE THREAD

**Issue by cobolbaby** [redpanda-data/console#248]
Deleting a topic is a very dangerous behavior. The system should be read-only by default. Only the logged-in user can have the delete permission.

**Comment by weeco** [comment]
> This option should indeed be configured in the production environment, but how to deal with some special scenarios where topics need to be deleted in the future?

I think one option would be to only enable this temporarily and switching it back to `enable=false` afterwards again.

I think I understand your point, but at the same time I believe from the maintainers' perspective as well as from a feature perspective it is desirable to have the proper solution with RBAC permissions. If we were to introduce feature switches the next person would love to have a feature switch for consumer groups (one for editing, one for deleting). The next person wants it to be scoped per topic or consumer group. And quickly we would end up reimplementing the pretty flexible permission system.

Does that make sense?

**Comment by vmercierfr** [comment]
We are in a similar situation, we are managing Kafka topics and schemas configurations as code, so we don't want to allow users to delete topics (or consumer groups since 1.4.0).

So we must switch to Kowl Business plan if we want to continue to use it.

Having a switch to just enable/disable this feature could be fair for the open-source version.
Can you add a switch to disable topic deletion in Kowl ?

**Comment by sw-tracker** [comment]
We are in the same boat, we actually wanted to deploy kwol open-source while we evaluated the business version. Now we have to deploy kafka-ui in the meantime which may make the developers not want to switch afterwards because it will require more effort. If we had this read-only flag (for topic creation and deletion) we would have deployed kwol open-source and try the get a business license in parallel (since developers would already be familiar with it and already using it).

Long story short, +1 for read-only flag. This was a deal breaker for us to use open-source kwol (or even trying it).

**Comment by weeco** [comment]
Hello everyone,
as of today we do not plan to add any access control functionality to the Console OSS version. This is however part of the Redpanda Enterprise platform. If you are interested in testing the enterprise version please reach out via https://redpanda.com/ .
